### PR TITLE
add example configuration using lazy.nvim

### PR DESCRIPTION
**Pull request by cmpadden**
Feel free to close w/ no hurt feelings. Had to do this anyway for my setup.

- Included an example using `lazy.nvim` -- feel free to close if you feel it's too noisy. Right now it's hidden behind a `<details>` toggle as to not expand the README too much.
- Also ran `stylua` on the _Using Packer_ example.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,43 +30,76 @@ Install using your favorite plugin manager:
...
+<details>
+<summary>Show configuration...</summary>
+
+```lua
+{
+    "PedramNavid/dbtpal",
+    dependencies = {
+        "nvim-lua/plenary.nvim",
+        "nvim-telescope/telescope.nvim",
+    },
```
Comment (PedramNavid) [inline README.md:80]: I had this to ensure it loads on the right filetype, did it work for you without this?
```suggestion
    },
    ft = { 'sql', 'yaml', 'markdown' },
```

File: README.md [hunk 2]
```diff
@@ -30,43 +30,76 @@ Install using your favorite plugin manager:
...
+        })
+
+        vim.keymap.set("n", "<leader>drf", dbt.run)
+        vim.keymap.set("n", "<leader>drp", dbt.run_all)
+        vim.keymap.set("n", "<leader>dtf", dbt.test)
+        vim.keymap.set("n", "<leader>dm", require("dbtpal.telescope").dbt_picker)
+
+        require("telescope").load_extension("dbtpal")
+    end,
+}
```
Comment (PedramNavid) [inline README.md:102]: Maybe adding some default keymaps too?
```suggestion
	keys = {
		{ 'n', '<LocalLeader>drf', function() require('dbtpal').run() end },
		{ 'n', '<LocalLeader>drp', function() require('dbtpal').run_all() end },
		{ 'n', '<LocalLeader>dtf', function() require('dbtpal').test() end },
		{ 'n', '<LocalLeader>dm', function() require('dbtpal.telescope').dbt_picker() end },
}
```

**Review by PedramNavid — COMMENTED** [review]
Just a few comments but looks good!

File: README.md (hunk 1 above)
Comment (cmpadden) [reply README.md:80]: Good catch, while it was loading after my initial key press, things like syntax highlighting weren't enabled on load. Added `ft`.

File: README.md (hunk 2 above)
Comment (cmpadden) [reply README.md:102]: Moved the keymaps to `keys`.  👍 
```
        keys = {
            { "<leader>drf", "<cmd>DbtRun<cr>" },
            { "<leader>drp", "<cmd>DbtRunAll<cr>" },
            { "<leader>dtf", "<cmd>DbtTest<cr>" },
            { "<leader>dm", "<cmd>lua require('dbtpal.telescope').dbt_picker()<cr>" },
        },
```

File: README.md [hunk 3]
```diff
@@ -30,43 +30,79 @@ Install using your favorite plugin manager:
...
+    },
+    config = function()
+        require("dbtpal").setup({
+            path_to_dbt = "dbt",
+            path_to_dbt_project = "",
+            path_to_dbt_profiles_dir = vim.fn.expand("~/.dbt"),
+            extended_path_search = true,
+            protect_compiled_files = true,
+        })
+        require("telescope").load_extension("dbtpal")
```
Comment (cmpadden) [inline README.md:100]: Side note, if we didn't have `require("telescope").load_extension("dbtpal")` we would be able to move all of the `setup` parameters to the _lazy.nvim_ opts parameter. Not sure how to get around loading the extension though.

Comment (PedramNavid) [reply README.md:100]: I think it's fine having it here right? Seems to work okay, not sure if there's any downsides.

Comment (cmpadden) [reply README.md:100]: Yeah, I think it's ok!

**Pull request merged**